### ISSUE TITLE
Android Gradle Sync is time-consuming

### ISSUE THREAD

**Issue by Nayuta403** [flutter/flutter#110559]
In our multiple projects upgrading from 2.0 to 2.10, we found that Android Gradle increased significantly.  Some projects went from 10s to 5 minutes. 

The discovery was introduced by this change: https://github.com/flutter/flutter/pull/83067. The [generateLockfiles](https://github.com/flutter/flutter/pull/83067/commits/3d47ad9fe75b8b14fa02f35bf74bf252c921f17c#diff-a6d94b09367a36221805cf68bf74922d8d74d89487a79a794e9b7813118aaf5aR106) task is introduced, and after we delete the task, the time is back to normal.

So I want to know: 
1. What's the worst thing that could happen if I cancel this task? 
2. Why is this task so time-consuming

**Comment by talisk** [comment]
+1

**Comment by Blankj** [comment]
2.0 sync
![image](https://user-images.githubusercontent.com/17978187/187339839-61be8223-ee65-40cb-abc4-312f59f263d4.png)
2.10 sync
![image](https://user-images.githubusercontent.com/17978187/187339872-5e3069c8-a98e-4456-8104-5cefe079c5c9.png)
I found 2.10 sync task execute many extra dependency task.

**Comment by Nayuta403** [comment]
cc @christopherfujino

**Comment by GaryQian** [comment]
Is this time increase only being observed recently or has it been happening since https://github.com/flutter/flutter/pull/83067 was introduced?

Also, does this occur on every run or just initial runs, perhaps after dependency changes?

1. Removing this task just disables dep locking, which means as packages upgrade, it might introduce breaking changes. In the short term, removing this task should not make much difference to your app.
2. We need to look into this further.

**Comment by Nayuta403** [comment]
@GaryQian  Thank you for response
> Is this time increase only being observed recently or has it been happening since https://github.com/flutter/flutter/pull/83067 was introduced?

Yes, especially when we remove [generateLockfiles](https://github.com/flutter/flutter/pull/83067/commits/3d47ad9fe75b8b14fa02f35bf74bf252c921f17c#diff-a6d94b09367a36221805cf68bf74922d8d74d89487a79a794e9b7813118aaf5aR106) task,  this problem does not exist. 

>  Removing this task just disables dep locking, which means as packages upgrade, it might introduce breaking changes.

Sorry, can you explain it in more detail? Will this cause my compilation to fail?

> We need to look into this further.

Yeah, thanks

**Comment by GaryQian** [comment]
> Sorry, can you explain it in more detail? Will this cause my compilation to fail?

No, not in the short term. In the future, you may have to fix your project as dependencies update. See https://docs.gradle.org/current/userguide/dependency_locking.html

Perhaps we don't have to call `gradlew {subproject}:dependencies --write-locks` on every build as this step generates artifacts that can be reused in the future.

**Comment by taibaiyinxing** [comment]
@GaryQian As [this link](https://docs.gradle.org/current/userguide/dependency_locking.html#lock_all_configurations_in_one_build_execution) said, there are two options to lock all configurations, can we use the second option instead of the first?

**Comment by reidbaker** [comment]
https://github.com/flutter/flutter/pull/112723 has a draft pr but tests fail. This might be a good starting point if someone wanted to take it over.
